local_pathfinding/node_mock_global_path.py

```diff
@@ -0,0 +1,220 @@
+        callback and check if the global path has been updated.
+
+    Attributes from subscribers:
+        gps (GPS): Data from the GPS sensor.
+
+    Attributes:
+        path_mod_tmstmp (Str): The modification timestamp of the csv file containing
+        the global path
+        file_path (Str): The filepath to the csv file containing the global path
+
+    Parameters:
+        pub_period_sec (Double): This parameter is used to set the period of the global path
+        callback function
+        global_path_filepath (Str): The filepath to the csv file containing the global path
+        interval_spacing (Double): The distance between waypoints on the global path
+    """
+
+    def __init__(self):
+        super().__init__(node_name="mock_global_path")
+
+        self.declare_parameters(
+            namespace="",
+            parameters=[
+                ("pub_period_sec", rclpy.Parameter.Type.DOUBLE),
+                ("global_path_filepath", rclpy.Parameter.Type.STRING),
+                ("interval_spacing", rclpy.Parameter.Type.DOUBLE),
+            ],
+        )
+
+        # Subscribers
+        self.gps_sub = self.create_subscription(
+            msg_type=GPS, topic="gps", callback=self.gps_callback, qos_profile=10
+        )
+
+        # Publishers
+        self.global_path_pub = self.create_publisher(
+            msg_type=Path, topic="global_path", qos_profile=10
+        )
+
+        # Path callback timer
+        pub_period_sec = self.get_parameter("pub_period_sec").get_parameter_value().double_value
+        self.get_logger().debug(f"Got parameter: {pub_period_sec=}")
+
+        self.global_path_timer = self.create_timer(
+            timer_period_sec=pub_period_sec,
+            callback=self.global_path_callback,
+        )
+
+        # Attributes
+        self.gps = MOCK_GPS  # TODO Remove when NET publishes GPS
+        self.path_mod_tmstmp = None
+        self.file_path = None
+
+    # Subscriber callbacks
+    def gps_callback(self, msg: GPS):
+        self.get_logger().debug(f"Received data from {self.gps_sub.topic}: {msg}")
+        self.gps = msg
+
+    # Timer callbacks
+    def global_path_callback(self):
+        """Check if the global path csv file has changed, on a regular time interval.
+        If it has changed, the new path is published.
+
+        Global path can be changed by modifying mock_global_path.csv or setting the
+        global_path_filepath parameter to a filepath to a new csv file.
+        """
+        if not self._all_subs_active():
+            self._log_inactive_subs_warning()
+
+        file_path = self.get_parameter("global_path_filepath")._value
+
+        # check when global path was changed last
+        path_mod_tmstmp = time.ctime(os.path.getmtime(file_path))
+
+        # Only publish if the path has changed
+        if (self.file_path != file_path) or (path_mod_tmstmp != self.path_mod_tmstmp):
```

Comment:
One thing I realized is that we may still want publish a new path if the current position from the GPS has changed a lot. This doesn't really matter right now since we're hardcoding it, but would be applicable once we actually read it from the GPS msg. Can implement in this PR or add a TODO comment